lib/src/firestore.dart

```diff
@@ -888,3 +887,110 @@ abstract class _Updatable {
     return callMethod(jsObject, "update", args);
   }
 }
+
+// Field values
+
+class _FieldValueDelete implements FieldValue {
+  @override
+  dynamic _jsify() {
+    return firestore_interop.FieldValue.delete();
+  }
+
+  @override
+  String toString() => 'FieldValue.delete()';
+}
+
+class _FieldValueServerTimestamp implements FieldValue {
+  @override
+  dynamic _jsify() {
+    return firestore_interop.FieldValue.serverTimestamp();
+  }
+
+  @override
+  String toString() => 'FieldValue.serverTimestamp()';
+}
+
+abstract class _FieldValueArray implements FieldValue {
+  final List elements;
+
+  _FieldValueArray(this.elements);
+}
+
+class _FieldValueArrayUnion extends _FieldValueArray {
+  _FieldValueArrayUnion(List elements) : super(elements);
+  @override
+  _jsify() {
+    // This uses var arg so cannot use js package
+    return callMethod(
+        firestore_interop.fieldValues, 'arrayUnion', jsifyList(elements));
+  }
+
+  @override
+  String toString() => 'FieldValue.arrayUnion($elements)';
+}
+
+class _FieldValueArrayRemove extends _FieldValueArray {
+  _FieldValueArrayRemove(List elements) : super(elements);
+  @override
+  _jsify() {
+    // This uses var arg so cannot use js package
+    return callMethod(
+        firestore_interop.fieldValues, 'arrayRemove', jsifyList(elements));
+  }
+
+  @override
+  String toString() => 'FieldValue.arrayRemove($elements)';
+}
+
+dynamic jsifyFieldValue(FieldValue fieldValue) => fieldValue._jsify();
```

Comment:
`jsifyFieldValue()` is explicitely hidden from the main `lib/firestore.dart` so it remains private:

```dart
export 'src/firestore.dart' hide jsifyFieldValue;
```

I did not want to export the `._jsify()` method but needed a way to access it from `utils.dart` in the main `jsify()` function:

https://github.com/tekartik-2/firebase-dart/blob/766a6c38ba12bde27cb3d8f81fce8acdea38a5a1/lib/src/utils.dart#L99-L101

I'm sure there could be better way but I did not want to change too much the existing. Let me know if you want me to arrange it differently (for example I would rather have a firestore specify jsify - since the existing is also used for storage and database where some data types are not supported). It is pretty much a copy of what I did for the node implementation: https://github.com/pulyaevskiy/firebase-admin-interop/commit/000276832e32f365adb39e2c280991fa6b4d79fe

Reply:
Ah! Just saw the hide. all good. thanks